### PR TITLE
Reorganize and update readme

### PR DESCRIPTION
**Pull request by faxu**
**Description**: Update readme; product description, reorganized sections for improved readability

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -143,32 +154,13 @@ This is particularly important when there are massive volumes of incoming data/s
  
 To deploy AI workloads to these edge devices and take advantage of hardware acceleration capabilities on the target device, see [these reference implementations](https://github.com/Azure-Samples/onnxruntime-iot-edge).
 
-### Local applications
-ONNX Runtime packages are published to PyPi and Nuget (see [Official Builds](#official-builds) and/or can be [built from source](./BUILD.md) for local application development. Find samples [here](https://github.com/microsoft/onnxruntime/tree/master/samples/c_cxx) using the C++ API. 
+### Client applications
```
[the remark this answers is not in the document]
Comment (faxu) [reply README.md]: Perhaps a bit early for that? @manashgoswami thoughts for including a mobile section?

File: README.md [hunk 2]
```diff
@@ -6,129 +6,140 @@
...
-* [Deploying ONNX Runtime](#deploying-onnx-runtime)
-* [Performance Tuning](#performance-tuning)
-
-**More Info**
-* [Technical Design Details](#technical-design-details)
-* [Extensibility Options](#extensibility-options)
+ONNX Runtime was designed with a focus on performance and scalability in order to support heavy workloads in high-scale production scenarios. It also has extensibility options for compatibility with emerging hardware developments.
 
-**[Data/Telemetry](#Data/Telemetry)**
+ONNX Runtime stays up to date with the ONNX standard and supports all operators from the ONNX v1.2+ spec with both forwards and backwards compatibility. Please refer to [this page](docs/Versioning.md) for ONNX opset compatibility details.
```
[the remark this answers is not in the document]
Comment (faxu) [reply README.md]: https://github.com/microsoft/onnxruntime/blob/master/docs/Versioning.md#forward-compatibility

That noted, we've seen this result in some problems so may limit support for this in the future. I'll remove the comment from the main readme.

File: README.md [hunk 3]
```diff
@@ -6,129 +6,140 @@
...
 
+#### Other
+* [Nuphar Model Compiler](./docs/execution_providers/Nuphar-ExecutionProvider.md)
 
-## Cross Platform
-ONNX Runtime is currently available for Linux, Windows, and Mac with Python, C#, C++, and C APIs. Please see [API documentation and package installation](#installation). 
+## Extensibility Options
+  * [Add a custom operator/kernel](docs/AddingCustomOp.md)
+  * [Add a new graph
+transform](include/onnxruntime/core/optimizer/graph_transformer.h)
```
[the remark this answers is not in the document]
Comment (faxu) [reply README.md]: which one? i used the lines to separate the topmost-level sections bc the size of the titles didn't seem distinctive enough. 
- functional overview
- installation
- usage
- contributions and feedback
- license

[the remark this answers is not in the document]
Comment (prasanthpul) [reply README.md]: there is a line break between line 74 and 75. it's actually the same line though

**Pull request merged**